apps/explorer/src/components/module/module-functions-interaction/FunctionExecutionResult.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getExecutionStatusError, getTransactionDigest } from '@mysten/sui.js';
+
+import { useTxEffectsObjectRefs } from './useTxEffectsObjectRefs';
+
+import type { SuiTransactionResponse } from '@mysten/sui.js';
+
+import Longtext from '~/components/longtext/Longtext';
+import { Banner } from '~/ui/Banner';
+
+type FunctionExecutionResultProps = {
+    result: SuiTransactionResponse | null;
+    error: string | false;
+    onClear: () => void;
+};
+
+export function FunctionExecutionResult({
+    error,
+    result,
+    onClear,
+}: FunctionExecutionResultProps) {
+    const adjError =
+        error || (result && getExecutionStatusError(result)) || null;
+    const variant = adjError ? 'error' : 'message';
+    const createdObjs = useTxEffectsObjectRefs(result, 'created');
+    const mutatedObjs = useTxEffectsObjectRefs(result, 'mutated');
+    const labelWithLinks = result
```

Comment:
I think this pattern is a little weird, this really could be defined in JSX instead, and I think that's a better place for it. I think it'd make more sense to make a component that renders the title + links, then you can not do this array building + mapping, it'd just be mapping directly in the JSX.

I imagine the usage would look something like this:

```jsx
<LinkGroup title="Created" links={createdObjs} />
<LinkGroup title="Updated" links={mutatedObjs} />
```

You could also just have it return null if there are no links which simplifies the display logic too.